src/v2/Apps/Auction2/Routes/Bid/Auction2BidRoute.tsx

```diff
@@ -0,0 +1,257 @@
+import { Button, Join, ModalDialog, Select, Spacer, Text } from "@artsy/palette"
+import { createFragmentContainer, graphql } from "react-relay"
+import { useRouter } from "v2/System/Router/useRouter"
+import { Auction2BidRoute_sale } from "v2/__generated__/Auction2BidRoute_sale.graphql"
+import { Auction2BidRoute_artwork } from "v2/__generated__/Auction2BidRoute_artwork.graphql"
+import { Auction2BidRoute_me } from "v2/__generated__/Auction2BidRoute_me.graphql"
+import { AuctionLotInfoFragmentContainer } from "./Components/AuctionLotInfo"
+import { dropWhile } from "lodash"
+import { Form, Formik } from "formik"
+import { OnSubmitValidationError } from "v2/Apps/Auction2/Components/Form/OnSubmitValidationError"
+import { PricingTransparency2QueryRenderer } from "./Components/PricingTransparency2"
+import { Match } from "found"
+import { useEffect } from "react"
+import { useSubmitBid } from "./useSubmitBid"
+import { AddressFormWithCreditCard } from "v2/Apps/Auction2/Components/Form/AddressFormWithCreditCard"
+import { ConditionsOfSaleCheckbox } from "v2/Apps/Auction2/Components/Form/ConditionsOfSaleCheckbox"
+import {
+  biddingValidationSchemas,
+  getSelectedBid,
+  initialValuesForBidding,
+} from "v2/Apps/Auction2/Components/Form/Utils"
+import { useAuctionTracking } from "v2/Apps/Auction2/Hooks/useAuctionTracking"
+import { CreditCardInputProvider } from "v2/Components/CreditCardInput"
+import { ErrorStatus } from "../../Components/Form/ErrorStatus"
+
+interface Auction2BidRouteProps {
+  sale: Auction2BidRoute_sale
+  artwork: Auction2BidRoute_artwork
+  me: Auction2BidRoute_me
+}
+
+const Auction2BidRoute: React.FC<Auction2BidRouteProps> = ({
+  sale,
+  artwork,
+  me,
+}) => {
+  const { match, router } = useRouter()
+  const { tracking } = useAuctionTracking()
+
+  const {
+    artworkSlug,
+    bidderID,
+    displayIncrements,
+    requiresCheckbox,
+    requiresPaymentInformation,
+    selectedBid,
+    validationSchema,
+  } = computeProps({ artwork, match, me })
```

Comment:
`non-blocking`: This pattern just highlights inadequacies in the GraphQL schema. Either to do with nullability or to do with computed fields which should be located there. I realize you've got other fish to fry right now; but just a comment.